gslib/tests/test_storage_url.py

```diff
@@ -135,3 +136,38 @@ def test_urls_are_mix_of_objects_and_buckets_is_true_for_a_mix(self):
   def test_urls_are_mix_of_objects_and_buckets_is_null_for_invalid(self):
     urls = list(map(storage_url.StorageUrlFromString, ['gs://b', 'f:o@o:o']))
     self.assertIsNone(storage_url.UrlsAreMixOfBucketsAndObjects(urls))
+
+  def test_urls_raise_error_if_mix_of_objects_and_buckets(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b1', 'gs://b/o']))
+    with self.assertRaisesRegex(
+            CommandException,
+            'Cannot operate on a mix of buckets and objects.'):
+      storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False)
+
+  def test_urls_raise_error_if_objects_and_buckets_are_mixed_different(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b/o', 'gs://b']))
+    with self.assertRaisesRegex(
+            CommandException,
+            'Cannot operate on a mix of buckets and objects.'):
+      storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False)
+
+  def test_accepts_mix_of_objects_and_buckets_if_recursion_requested(self):
+    # No error raised
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b1', 'gs://b/o']))
+    storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=True)
+  
+  def test_not_raising_error_if_mix_objects_with_recursion(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b/o', 'gs://b/p']))
+    self.assertFalse(storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=True))
+
+  def test_not_raising_error_if_mix_objects_without_recursion(self):
```

Comment:
Would call this "multiple_objects" instead of "mix" b/c we're using "mix" to refer to a combination of buckets and objects in all the other tests. Ditto for the buckets test below.